hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/HollowScreening.java

```diff
@@ -0,0 +1,183 @@
+
+/**
+ * A utility class for {@link Expansion} and {@link Rationalization} that, given a list of {@link TransactionSignature}
+ * and infrastructure for looking up accounts, is able to construct a list of {@link PendingCompletion} which may be
+ * performed in the given {@link SwirldsTxnAccessor}, if all key/sig activation validations pass.
+ *
+ * <p>Also exposes methods that, given {@link JWildcardECDSAKey}s as input, replace the {@link JWildcardECDSAKey}/s with their
+ * corresponding {@link JECDSASecp256k1Key}, if such was present in the list of parsed {@link TransactionSignature}s.
+ * This process is also referred to as "de-hollowing". The {@code payerReqSig} and {@code otherReqSigs} contained
+ * in the {@link RationalizedSigMeta} must be free of
+ * {@link JWildcardECDSAKey}s, so that subsequent key activation tests pass, so this de-hollow process is essential.
+ */
+public class HollowScreening {
+
+    static boolean atLeastOneWildcardECDSAKeyIn(@NonNull final JKey payerKey, @NonNull List<JKey> otherPartyKeys) {
+        if (payerKey.hasWildcardECDSAKey()) {
+            return true;
+        }
+        for (final var otherReqKey : otherPartyKeys) {
+            if (otherReqKey.hasWildcardECDSAKey()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Encapsulates the result of a hollow screening.
+     *
+     * @param pendingCompletions a list of {@link PendingCompletion} if there the hollow screening found possible
+     *                           hollow account completions. {@code null} otherwise
+     * @param replacedPayerKey a {@link JECDSASecp256k1Key} if payer key was {@link JWildcardECDSAKey} and a corresponding
+     * {@link JECDSASecp256k1Key} was found in the txn sigs. {@code null} otherwise
+     * @param replacedOtherKeys a list of all other req keys, where possible {@link JWildcardECDSAKey} -> {@link JECDSASecp256k1Key}
+     *                          conversions were made; {@code null} if no conversions were made
+     */
+    record HollowScreenResult(
+            @Nullable List<PendingCompletion> pendingCompletions,
+            @Nullable JKey replacedPayerKey,
+            @Nullable List<JKey> replacedOtherKeys) {}
+
+    /**
+     * <strong>NOTE</strong>- should be called only if {@link HollowScreening#atLeastOneWildcardECDSAKeyIn} for the same
+     * {@code payerKey} and {@code otherKeys} returned true
+     *
+     * For each {@link JWildcardECDSAKey} in {@code payerKey} and {@code otherKeys}, tries to find its corresponding
+     * {@link JECDSASecp256k1Key} from the {@code txnSigs}, and also construct a {@link PendingCompletion} if a hollow
+     * account is linked to the evm address from the {@link JWildcardECDSAKey};
+     *
+     * <p>Note that this method tries to replace {@link JWildcardECDSAKey}s present in {@code otherKeys}
+     * even if no pending completions are present. That's because a CryptoCreate with an evm address alias, derived from
```

Comment:
```
... That's because a CryptoCreate with an evm address alias, derived from
     * a key, different than the key being set for the account, also adds a {@link JWildcardECDSAKey} to the {@code otherKeys}
```
Where _is_ the alias-derived `JWildcardECDSAKey` being added to `otherKeys`, actually? 😳 (I expected to find it in [`SigRequirements.cryptoCreate()`](https://github.com/hashgraph/hedera-services/blob/04983-new-hollow-finalization/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/order/SigRequirements.java#L733), but it doesn't seem to be there.)

Reply:
😆 It's in the [CryptoCreate with an alias derived from another key](https://github.com/hashgraph/hedera-services/pull/5011/files#diff-be69893f0a4a056ac0f9f260a694192066908f8dcce404218664ef02b61bdd6fR740-R760) PR. 

Kind of jumped the gun here, wanted to tie in everything at once, and then just base the other PR on the changes here

Reply:
Ah! Ok that makes sense. 😁